excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/parser/NoCConfigParser.java

```diff
@@ -0,0 +1,106 @@
+package uk.gov.hmcts.ccd.definition.store.excel.parser;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import uk.gov.hmcts.ccd.definition.store.domain.validation.ValidationError;
+import uk.gov.hmcts.ccd.definition.store.domain.validation.ValidationException;
+import uk.gov.hmcts.ccd.definition.store.domain.validation.ValidationResult;
+import uk.gov.hmcts.ccd.definition.store.excel.endpoint.exception.MapperException;
+import uk.gov.hmcts.ccd.definition.store.excel.parser.model.DefinitionDataItem;
+import uk.gov.hmcts.ccd.definition.store.excel.parser.model.DefinitionSheet;
+import uk.gov.hmcts.ccd.definition.store.excel.util.mapper.ColumnName;
+import uk.gov.hmcts.ccd.definition.store.excel.util.mapper.SheetName;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.CaseTypeEntity;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.NoCConfigEntity;
+
+import static java.util.Optional.ofNullable;
+
+@Slf4j
+public class NoCConfigParser {
+
+    private ParseContext parseContext;
+
+
+    public NoCConfigParser(ParseContext parseContext) {
+        this.parseContext = parseContext;
+    }
+
+    public Map<String, List<NoCConfigEntity>> parse(Map<String, DefinitionSheet> definitionSheets) {
+        Map<String, List<DefinitionDataItem>> nocConfigsCaseType = ofNullable(definitionSheets.get(SheetName.NOC_CONFIG.getName()))
+            .map(DefinitionSheet::groupDataItemsByCaseType)
+            .orElse(Collections.emptyMap());
+
+        checkInvalidCaseTypeIds(nocConfigsCaseType);
+        Map<String, List<NoCConfigEntity>> caseTypeNoCConfigEntities = new HashMap<>();
+        ValidationResult validationResult = new ValidationResult();
+        parseContext.getCaseTypes()
```

Comment:
Could we separate the validation logic from parsing ?  may be the parser can accumulate validation errors and throw them at the end

Reply:
This ticket has only 2 columns.  It is ok leaving the validation as simple method. May we could create one validator if there are a lot of fields to validate .